### ISSUE TITLE
Git exporter auth is too eager

### ISSUE THREAD

**Issue by paour** [WeblateOrg/weblate#1420]
I wanted to export my repo to resolve a conflict, but I didn't want to leave it open, yet I didn't want to activate per project access control, so I added an Apache basic auth filter.

### Steps to reproduce
1. enable `weblate.gitexport` and perform `./manage.py migrate`
2. on the apache front-end, add the following config below the existing Weblate directives to allow access to `wsgi.py`
```
<Location "/git">
    AuthName "Please login to access Git"
    AuthType Basic
    AuthUserFile "/etc/apache2/passwd"
    Require valid-user
</Location>
```
3. create the password file with an arbitrary password

### Actual behaviour
Via a browser or curl, I was able to verify that connecting to the rest of Weblate was still open, but that anything below git was secured. Using curl to connect to `/git/project_id/component_id` also worked and showed a redirection to the project web page `/project/project_id/component_id`.

However, trying to `git clone https://host/git/project_id/component_id` gave auth failures every time.

### Expected behaviour
The git clone should have worked (and did if I removed the Apache authentication directives).

### What's happening
Using the apache detailed logging facility, I was able to determine that while my Apache authentication directives matched the auth header git was sending, a secondary authentication challenge failed. I was puzzled, because I was getting two different auth realms: my `Please login to access Git` prompt, and later, just `Git`.

On a hunch, I changed my password to match the Weblate user key, and presto! it worked.

Long story short: the GitExport module should ignore auth headers when ACLs are disabled, rather than triggering whenever an auth header is present.

**Comment by nijel** [comment]
Authentication to Git repos will be hopefully be addressed at Weblate level in next release, see https://github.com/WeblateOrg/weblate/issues/1336.

But indeed if Weblate won't provide this option, it should allow to skip authentication.

**Comment by nijel** [comment]
Also adding `WSGIPassAuthorization off` to your `<Location>` entry might fix this for you (haven't tested, I'm not sure if it can be applied at this level).

**Comment by nijel** [comment]
Did you try the suggested workaround if that worked?

Anyway in upcoming release, there is permission for Git access, so you can set it only to yourself (currently in admin interface only, but it will be in Weblate as well).

**Comment by paour** [comment]
> Did you try the suggested workaround if that worked?

I haven't, since I had another work-around. I'll stay with my current work-around until the new version, if that's OK :-)
